Client/Classes/Dialogs/ProfessionDialog.sqx

```diff
@@ -95,6 +160,100 @@ namespace Intrusion.Client
 			};
 		};
 		
+ 		// Change Image and text when a new selection is made in the listbox.
+		// Called by the .hpp-dialog
+		public static method OnListSelectChanged {
+			private [
+				"_profession" as Profession, 
+				"_vehicles" as ProfessionVehicleClassNamesInfo, 
+				"_technology" as TechnologyInfo,
+				"_vehiclesConfig" as IProfessionVehiclesConfig,
+				"_vehicleInfo" as ProfessionVehicleInfo,
+				"_className" as String,
+				"_thisData" as String,
+				"_selectedIndex" as Scalar
+			];
+			private _model = "";
+			private _vehicleNames = []; 
+			private _ctrlVehicleModel = (findDisplay 2999) displayCtrl 2230;
+			private _ctrlDescription = (findDisplay 2999) displayCtrl 2215;
+			//_profession = call _self.GetSelectedProfessionTypeFromCombo;
+			_profession = [call _self.GetSelectedProfessionTypeFromCombo] call _mProfessionConfig.GetProfession;
+			_vehicles = [side player, _profession.Type] call gCommonConfig.VehicleClassNames.GetProVehiclesClassNameInfo;
```

Comment:
Globalerna (gClient, gCommon, gServer) får inte användas någon annanstans än i initfilerna och MissionEngine (som är enda undantaget). Skicka istället in behövd config i en klassvariabel - _mVehicleClassNamesConfig.